pageserver/src/http/routes.rs

```diff
@@ -495,13 +495,39 @@ async fn get_lsn_by_timestamp_handler(
     let timeline = active_timeline_of_active_tenant(tenant_id, timeline_id).await?;
     let result = timeline.find_lsn_for_timestamp(timestamp_pg, &ctx).await?;
 
-    let result = match result {
-        LsnForTimestamp::Present(lsn) => format!("{lsn}"),
-        LsnForTimestamp::Future(_lsn) => "future".into(),
-        LsnForTimestamp::Past(_lsn) => "past".into(),
-        LsnForTimestamp::NoData(_lsn) => "nodata".into(),
-    };
-    json_response(StatusCode::OK, result)
+    if request
+        .headers()
+        .get(header::ACCEPT)
+        .map(|v| v == "application/json")
+        .unwrap_or_default()
+    {
+        #[derive(serde::Serialize)]
+        struct Result {
+            lsn: String,
```

Comment:
Why string instead of just Lsn?

Before the (unmerged ext contributor PR) it will require serde_with and serde_as.

Reply:
A string makes it clear which serialization format is chosen. Lsn is a tuple struct with a field named 0 that contains an integer. I don't want it to be serialized as `{ "0": 12345678 }` however. Using `serde_with`/`serde_as` for this feels like overengineering to me. The simplest would be to just use the `json` macro...

Reply:
Addressed in a change, I use `serde_as` now.